core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -360,8 +360,11 @@ static uint64_t mattr_to_desc(unsigned level, uint32_t attr)
 	if (a & TEE_MATTR_UW)
 		a |= TEE_MATTR_PW;
 
-	if (IS_ENABLED(CFG_CORE_BTI) && (a & TEE_MATTR_PX))
+#ifdef ARM64
+	if (core_is_bti_enabled() && IS_ENABLED(CFG_CORE_BTI) &&
```

Comment:
Perhaps we can swap these two to let the compiler skip calling `core_is_bti_enabled()` in case `CFG_CORE_BTI=n`.

Reply:
Sure. In face as I examined it more, it is better to move this check to line 385 where GP bit is set in the descriptor. I will make the change.